addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -486,6 +486,8 @@ def action_refuse(self):
     def _check_approval_update(self, state):
         """ Check if target state is achievable. """
         current_employee = self.env['hr.employee'].search([('user_id', '=', self.env.uid)], limit=1)
+        if not current_employee:
```

Comment:
I'm not sure I understand this anymore. Shouldn't it be:

```python
if not self.mapped('employee_id'):
    return
```
to check if there is not employee set on the allocation?